.github/workflows/run_examples.yml

```diff
@@ -1,6 +1,9 @@
 name: Run Examples Tests
 on:
   workflow_dispatch:
+  push:
+    branches:
+      - master
```

Comment:
Consider adding a comment explaining why example tests are run on master branch pushes. This helps future maintainers understand the CI/CD pipeline design decisions.